packages/block-editor/src/components/provider/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { flow } from 'lodash';
+
+/**
+ * WordPress Dependencies
+ */
+import { createElement, Component } from '@wordpress/element';
+import { DropZoneProvider, SlotFillProvider } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+class BlockEditorProvider extends Component {
+	constructor( props ) {
+		super( ...arguments );
+		props.updateEditorSettings( props.settings );
+		props.initBlocks( props.value );
+		this.persistedValue = props.select( 'core/block-editor' ).getBlocks();
```

Comment:
We can't because the props the `initBlocks` call right before changes this prop and we need to save the persisted value before the component rerenders to ensure we don't have an infinite loop.

Reply:
> Can we use `props.blocks` here instead of calling `select()`? Or is this is a workaround? If so, a comment would be helpful.

> We can't because the props the `initBlocks` call right before changes this prop and we need to save the persisted value before the component rerenders to ensure we don't have an infinite loop.

But it seems that, if we had called on the preceding line `initBlocks` using `props.value`, that we should expect the following to be true:

```js
props.value === props.select( 'core/block-editor' ).getBlocks()
```

It's not the case though, due to all the transforms which occur between a blocks value being [pushed into the `block-editor` state](https://github.com/WordPress/gutenberg/blob/d4832c4734685d3bb8fa2555f067f5ee0ba27913/packages/block-editor/src/store/reducer.js#L219-L253) and [pulled back out with `getBlocks`](https://github.com/WordPress/gutenberg/blob/d4832c4734685d3bb8fa2555f067f5ee0ba27913/packages/block-editor/src/store/selectors.js#L203-L223).

Other options might be to avoid the transforms (not very plausible), or store a reference to the initializing blocks value and using it as a short-cut for `getBlocks` when we can be confident that no changes have been made to its state.

At the very least, the request for inline code comment was not implemented, and I'd second it.

Reply:
Also, what would be the difference between `props.blocks` and `props.select( 'core/block-editor' ).getBlocks()` at this point in the component lifecycle?